### PR TITLE
bump to libfaketime 0.9.10

### LINKED ISSUE #63: Upgrade to libfaketime 0.9.10 to unbreak OSX Monterey support
`libfaketime` recently [updated to 0.9.10](https://github.com/wolfcw/libfaketime/releases/tag/v0.9.10), partly to [fix an issue](https://github.com/wolfcw/libfaketime/issues/357) caused by upgrades to OSX Monterey. We're working around it currently using `freezegun`, but would prefer to drop that and keep on with `python-libfaketime`.

### LINKED ISSUE #64: libfaketime build issues on gcc 11 and glibc 2.34
This patch resolves the issue for me https://github.com/wolfcw/libfaketime/commit/0d964363a4352b7ee358f523c3416d07378d4583

```
  make[1]: Entering directory '/build/libfaketime-2.0.0/libfaketime/vendor/libfaketime/src'
  gcc -o libfaketime.o -c -std=gnu99 -Wall -Wextra -Werror -Wno-nonnull-compare -fPIC -DPREFIX='"'/usr/local'"' -DLIBDIRNAME='"'/lib/faketime'"'   libfaketime.c
  In file included from /nix/store/1dlzfarjz8g5hx8zhcvsglbp8bqing4r-glibc-2.34-210-dev/include/string.h:519,
                   from libfaketime.c:36:
  In function ‘strncpy’,
      inlined from ‘fake_clock_gettime.part.0’ at libfaketime.c:2095:5:
  /nix/store/1dlzfarjz8g5hx8zhcvsglbp8bqing4r-glibc-2.34-210-dev/include/bits/string_fortified.h:95:10: error: ‘__builtin_strncpy’ specified bound 256 equals destination size [-Werror=stringop-truncation]
     95 |   return __builtin___strncpy_chk (__dest, __src, __len,
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     96 |                                   __glibc_objsize (__dest));
        |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  cc1: all warnings being treated as errors
```

### PR DESCRIPTION
**Pull request by azmeuk**
This PR brings the following things

- update to libfaketime 0.9.10
- remove the pre_sierra submodule, as I suspect this starts to refer to very old versions?
- use `FAKETIME_FMT` environment var instead of patching libfaketime
- deactivate libfaketime cache with the `FAKETIME_NO_CACHE` env var instead of patching libfaketime
- use make/gcc compilation options instead of patch the libfaketime Makefile
- use libfaketime upstream as git submodule, so no fork is needed
- run GHA on macos 12, 13 and 14

fixes #63 
fixes #64

### CODE REVIEW

File: libfaketime/__init__.py [hunk 1]
```diff
@@ -208,7 +204,13 @@ def __exit__(self, *exc):
                 os.environ['FAKETIME'] = self._prev_spec
             else:
                 del os.environ['FAKETIME']
-                end_callback(self)
```
Comment (simon-weber) [inline libfaketime/__init__.py:211]: huh, looks like this was a bug before, yeah? it should have been one level out?

**Comment by simon-weber** [comment]
This looks fantastic; thanks! It'll be so much easier to upgrade libfaketime now.

File: libfaketime/__init__.py (hunk 1 above)
Comment (azmeuk) [reply libfaketime/__init__.py:211]: Yes I suppose, else that would imply that the end callback is only called if there was not a previous `FAKETIME` environment var defined, and I am not sure how it makes sense.

Comment (azmeuk) [reply libfaketime/__init__.py:211]: I can do a fix for this in a separate PR if you prefer.

Comment (simon-weber) [reply libfaketime/__init__.py:211]: looks like it's been fixed. I just wanted to double check it.

**Pull request merged**